nimbus/src/lib.rs

```diff
@@ -80,7 +82,23 @@ impl NimbusClient {
             .map(|e| e.branch_slug.clone()))
     }
 
+    pub fn get_global_user_participation(&self) -> Result<Option<bool>> {
+        self.db()?.get(StoreId::Meta, DB_KEY_GLOBAL_USER_PARTICIPATION)
+    }
+
+    pub fn set_global_user_participation(&self, flag: bool) -> Result<()> {
+        self.db()?.put(StoreId::Meta, DB_KEY_GLOBAL_USER_PARTICIPATION, &flag)
+    }
+
+    fn is_enabled(&self) -> Result<bool> {
+        let user_choice = self.get_global_user_participation()?;
+        Ok(user_choice.unwrap_or(DEFAULT_GLOBAL_USER_PARTICIPATION))
+    }
```

Comment:
Unpacking whether the user has opted in or out, and how that actually affects the library is done in a private method so we can use it.

I'm not sure of the caching properties of `rkv`, so unsure whether it's worth storing a property on this `NimbusClient` struct itself.